evalml/pipelines/component_graph.py

```diff
@@ -120,7 +121,7 @@ def fit(self, X, y):
             y (ww.DataColumn, pd.Series): The target training data of length [n_samples]
         """
         X = infer_feature_types(X)
-        X = _convert_woodwork_types_wrapper(X.to_dataframe())
```

Comment:
First bug: there's no need to convert to pandas here. That's all handled in `_compute_features`.

Symptom: calls to pipeline `fit` always use whatever types woodwork infers for all the columns, overriding all user-specified values.

Explanation: the first thing we do in `_compute_features` is call `X = infer_feature_types(X)`. So, by converting to pandas here and then immediately converting back to woodwork first thing in `_compute_features`, we were effectively resetting the input to whatever woodwork infers by default.

Reply:
@dsherry this is very interesting and I wonder if it's part of what I was seeing in my investigation with Chris.